inc/database/storage.php

```diff
@@ -181,7 +189,10 @@ private function get_target_orders( $object_id, $type, $source, $target ) {
 		global $wpdb;
 
 		$items = $wpdb->get_results( $wpdb->prepare(
-			"SELECT `$target` AS `id`, `order_$target` AS `order` FROM {$wpdb->mb_relationships} WHERE `$source` = %d AND `type` = %s",
+			"SELECT `$target` AS `id`, `order_$target` AS `order` FROM {$wpdb->mb_relationships} WHERE `$source` = %d AND `type` = %s
```

Comment:
Do we need to set **both** source and target orders here? I don't see any reason for this.

Reply:
Actually, we are not setting the order of both extremes, but we are getting the unified order regardless of the extreme for which the relationship was set.

Previously if you had set the relationships from A to B, A to C, A to D, you had no problem.
But if you had set for example C to A, you were always getting the order of the `target` when in that case it corresponded to the opposite extreme.

Here is an example of what we are getting now from the database
![image](https://user-images.githubusercontent.com/11416255/83692824-19eb0c00-a5cb-11ea-8ac1-30407d3ca617.png)

Reply:
I think the code should works as it's too safe to get the order of both sides (`order_from` and `order_to`).

My question should have been clearer: since the `get_target_orders` is used to get the order from only one side to use in the `update` method. Is it redundant to get the orders for both sides?

Reply:
I think the code should works as it's too safe to get the order of both sides (`order_from` and `order_to`).

My question should have been clearer: since the `get_target_orders` is used to get the order from only one side to use in the `update` method. Is it redundant to get the orders for both sides?

Reply:
It's not redundant since the update method previously assumed that the order it should get was always the `order_$target`. It assumed that the post you were updating was always in the `$target` column of the mb_relationships table. And that is not always the case with reciprocal relationships.

That is why we now get the order (`order_$target` OR `order_$source`) depending on which column the post we are updating is in.
In this way it does not matter from which end we update a reciprocal relationship or from where the relationship was set, we will be updating the order in a consistent way.

This is not a problem when the relationships are not reciprocal because the `get_target` and `get_source` functions return the correct side since the `post_type` is different.

I hope I have explained myself better.